File: app/src/main/java/com/example/mobv2/callbacks/SetAddressCallback.java
package com.example.mobv2.callbacks;

import android.util.Log;

import com.example.mobv2.serverapi.MOBServerAPI;
import com.example.mobv2.ui.activities.MainActivity;
import com.google.gson.internal.LinkedTreeMap;

public class SetAddressCallback implements MOBServerAPI.MOBAPICallback
{
    private final MainActivity mainActivity;

    public SetAddressCallback(MainActivity mainActivity)
    {
        this.mainActivity = mainActivity;
    }

    @Override
    public void funcOk(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());

        var response = (LinkedTreeMap<String, String>) obj.get("response");

        MainActivity.token = response.get("token");
    }

    @Override
    public void funcBad(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());
    }

    @Override
    public void fail(Throwable obj)
    {
        Log.v("DEBUG", obj.toString());
    }
}

File: app/src/main/java/com/example/mobv2/callbacks/SetAddressCallback.kt
package com.example.mobv2.callbacks

import android.util.Log
import com.example.mobv2.ui.activities.MainActivity
import com.google.gson.internal.LinkedTreeMap
import serverapi.MOBServerAPI.MOBAPICallback

class SetAddressCallback(private val mainActivity: MainActivity) : MOBAPICallback
{
    override fun funcOk(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
        val response = obj["response"] as LinkedTreeMap<String, String>?

        MainActivity.token = response!!["token"]
    }

    override fun funcBad(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
    }

    override fun fail(obj: Throwable)
    {
        Log.v("DEBUG", obj.toString())
    }
}